src/apiStrategy.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+// Requirements
+
+const EventEmitter = require('events');
+const randomstring = require('randomstring');
+const path = require('path');
+const Rest = require('./transports/Rest');
+const WS = require('./transports/WS');
+
+
+// API Strategy
+
+/**
+* @event onMessage
+*/
+class APIStrategy extends EventEmitter {
+
+    /**
+     * APIStrategy
+     */
+    constructor(urls) {
+        super();
+
+        const SelectedAPI = this.getType(urls.mainServiceURL);
```

Comment:
const SelectedAPI = this.getType(urls.mainServiceURL);

Please, change name to better one. For example:

const SelectedTransport = this.getTransport(urls.mainServiceURL);